crypto/crypto.go

```diff
@@ -257,11 +267,31 @@ func DecodePublicKeyPEM(sigAlgo SignatureAlgorithm, s string) (PublicKey, error)
 	}
 
 	goPublicKey := publicKey.(*ecdsa.PublicKey)
-
-	rawPublicKey := append(
-		goPublicKey.X.Bytes(),
-		goPublicKey.Y.Bytes()...,
-	)
+	xBytes := goPublicKey.X.Bytes()
```

Comment:
using `ParsePKIXPublicKey` only works in the case of curve P256. We will need to write a local `ParsePKIXPublicKey` function that supports secp256k1. I can help writing it.